### ISSUE TITLE
Function returning a promise

### ISSUE THREAD

**Issue by kennedyrs** [masasron/adonis-throttle#18]
When using Redis, the function below returns a promise to the count variable

```js
count() {
        let count = this.store.get(this.key)
        if (typeof count === 'undefined') {
            return 0
        }
        return count
    }
```

**Comment by AndrewJo** [comment]
Hey @kennedyrs, thanks for reporting the bug. Redis driver uses promises because network I/O to Redis shouldn't block the Node.js event loop. This is not a problem for Memory driver since we can directly access the memory synchronously without blocking the event loop.

In fact, any driver written for external caches such as Redis, Memcached, etc. will have to be promises.

@masasron is it alright if I promisify all of the `this.store` methods? Memory driver will just resolve the promise right away (e.g. wrapped in `Promise.resolve()`) but it should make accessing the store much more consistent.

**Comment by masasron** [comment]
@AndrewJo yes I think that would be a good idea, but since this is a breaking change for the memory driver we need to release this as a minor release.

**Comment by ruanitto** [comment]
Edit Redis driver, line 54:

```javascript
  get(key) {
    // return this.redis.get(key)
    this.redis.get(key, (err, result) => {
      if (err) {
        throw err
      } else {
        return result
      }
    })
  }
```
